tests/tests/test-xcm/test-xcmv3-new-instructions.ts

```diff
@@ -0,0 +1,382 @@
+        .buy_execution()
+        .expect_asset()
+        .as_v3();
+
+      // Mock the reception of the xcm message
+      await injectHrmpMessage(context, paraId, {
+        type: "XcmVersionedXcm",
+        payload: xcmMessage,
+      } as RawXcmMessage);
+      await context.createBlock();
+
+      // Search for ExpectationFalse error
+      const records = (await context.polkadotApi.query.system.events()) as any;
+      const events = records.filter(
+        ({ event }) => event.section == "xcmpQueue" && event.method == "Fail"
+      );
+      expect(events).to.have.lengthOf(1);
+      expect(events[0].toHuman().event.data.error).equals("ExpectationFalse");
+    });
+
+    it("Should execute ExpectOrigin (ExpectationFalse)", async function () {
+      const xcmMessage = new XcmFragment(dotAsset)
+        .withdraw_asset()
+        .buy_execution()
+        .expect_origin()
+        .as_v3();
+
+      // Mock the reception of the xcm message
+      await injectHrmpMessage(context, paraId, {
+        type: "XcmVersionedXcm",
+        payload: xcmMessage,
+      } as RawXcmMessage);
+      await context.createBlock();
+
+      // Search for ExpectationFalse error
+      const records = (await context.polkadotApi.query.system.events()) as any;
+      const events = records.filter(
+        ({ event }) => event.section == "xcmpQueue" && event.method == "Fail"
+      );
+      expect(events).to.have.lengthOf(1);
+      expect(events[0].toHuman().event.data.error).equals("ExpectationFalse");
+    });
+
+    it("Should execute ExpectError (ExpectationFalse)", async function () {
+      const xcmMessage = new XcmFragment(dotAsset)
+        .withdraw_asset()
+        .buy_execution()
+        .expect_error()
+        .as_v3();
+
+      // Mock the reception of the xcm message
+      await injectHrmpMessage(context, paraId, {
+        type: "XcmVersionedXcm",
+        payload: xcmMessage,
+      } as RawXcmMessage);
+      await context.createBlock();
+
+      // Search for ExpectationFalse error
+      const records = (await context.polkadotApi.query.system.events()) as any;
+      const events = records.filter(
+        ({ event }) => event.section == "xcmpQueue" && event.method == "Fail"
+      );
+      expect(events).to.have.lengthOf(1);
+      expect(events[0].toHuman().event.data.error).equals("ExpectationFalse");
+    });
+
+    it("Should execute QueryPallet (Unroutable)", async function () {
+      const xcmMessage = new XcmFragment(dotAsset)
+        .withdraw_asset()
+        .buy_execution()
+        .query_pallet(1002)
+        .as_v3();
+
+      // Mock the reception of the xcm message
+      await injectHrmpMessage(context, paraId, {
+        type: "XcmVersionedXcm",
+        payload: xcmMessage,
+      } as RawXcmMessage);
+      await context.createBlock();
+
+      // Search for Unroutable error
+      const records = (await context.polkadotApi.query.system.events()) as any;
+      const events = records.filter(
```

Comment:
same comment as in the `tests/tests/test-xcm/test-xcmv3-max-weight-instructions.ts`